src/core/StringX.h

```diff
@@ -28,6 +28,8 @@
 #include "../os/typedefs.h"
 #include "./PwsPlatform.h"
 
+extern void trashMemory(void *buffer, size_t length);
```

Comment:
Don't see how an #include in the cpp file can cause a circular dependency, but if that's the case please add a comment explaining this above the extern statement.